CA-322784 Fix varstored-guard logging is verbose

lippirk opened the pull request:
Increase log level to info, but add command line
switch to allow us to change it to debug if
necessary.

Also log to Local5 (xensource.log) rather than
Daemon (daemon.log).

Signed-off-by: lippirk <ben.anson@citrix.com>

edwintorok commented on src/dune:
```diff
@@ -7,7 +7,8 @@
 	xapi-idl.varstore.privileged
-	xen-api-client-lwt)
+	xen-api-client-lwt
+	cmdliner)
```
We try to keep this list sorted, please move this up.

edwintorok commented on src/main.ml:
```diff
@@ -147,12 +147,25 @@ let make_message_switch_server () =
+  let info = Cmdliner.(Term.info "varstored-guard" ~exits:Term.default_exits) in
+  let log_level =
+    let doc = "Syslog level. E.g. debug, info etc." in
+    Cmdliner.(Term.(const Syslog.level_of_string $ Arg.(value & opt string "info" & info ["log-level"] ~docv:"LEVEL" ~doc))) in
```
You could use an `'a Cmdliner.Arg.conv` here:
```
let level : Syslog.level Cmdliner.Arg.conv = 
   let parse s = try `Ok (Syslog.level_of_string s) with _ -> `Error "Unknown level" in
   let print ppf level = Format.pp_print_string ppf (Syslog.string_of_level level) in
   parse, print in
Cmdliner.(Term.(const main $ Arg.(value & opt level Syslog.Info & info ["log-level"] ~docv:"LEVEL" ~doc
```

edwintorok commented on src/main.ml:
```diff
@@ -147,12 +147,25 @@ let make_message_switch_server () =
+
+  Debug.set_facility Syslog.Local5;
+  match get_log_level () with
+  | `Ok log_level -> Debug.set_level log_level; main ()
```
Usually we let cmdliner drive the program from here, we don't query individual arguments, like this:
```
Term.exit @@ Term.eval cmd
```

edwintorok commented on src/main.ml:
```diff
@@ -147,12 +147,25 @@ let make_message_switch_server () =
   | `Error (`Msg m) ->
     Lwt.fail_with (Printf.sprintf "Failed to listen on message-switch queue: %s" m)
 
+let get_log_level () =
```
We usually define a command here, so perhaps call this `let cmd = `, and move it below such that `main` is already in scope.

edwintorok commented on src/main.ml:
```diff
@@ -147,12 +147,25 @@ let make_message_switch_server () =
-    );
-  let () = Lwt_main.run @@ make_message_switch_server () in
-  D.debug "Exiting varstored-guard"
+  let main () =
```
main will have to take a level argument

Commit:
maintenance: setup travis

Signed-off-by: lippirk <ben.anson@citrix.com>

Commit:
CA-322784 Fix varstored-guard logging is verbose

Increase log level to info, but add command line
switch to allow us to change it to debug if
necessary.

Also log to Local5 (xensource.log) rather than
Daemon (daemon.log).

Signed-off-by: lippirk <ben.anson@citrix.com>

edwintorok reviewed (approved):
Thanks, LGTM
